pode/pode.py

```diff
@@ -1,1464 +1,785 @@
+def order_by_sites(points: Multipoint,
+                   site_location: Point) -> List[Point]:
     """
-    For a graph (N1, N2, ..., Nm) and its node Nj
-    returns all Nj's successor nodes (Nj+1, Nj+2, ..., Nm).
-    Note that this definition is different
-    from the definition of successors for directed graphs.
+    Orders vertices of the convex polygon and the sites in a
+    counterclockwise manner so that the last vertex would be a site.
+    :param points: convex polygon's vertices and sites
+    :param site_location: site that will be the last vertex
+    :return: ordered union of polygon vertices and sites
     """
-    not_a_node = partial(ne, node)
-    node_and_successors = dropwhile(not_a_node, graph.nodes)
-    yield from islice(node_and_successors, 1, None)
+    centroid_ = centroid(points)
 
+    def angle(point: Point) -> float:
+        return atan2(point.y - centroid_.y, point.x - centroid_.x)
 
-def update_graph(polygon: Polygon,
-                 sites: SitesType,
-                 edge: Optional[LineString],
-                 neighbor: Polygon,
-                 graph: nx.Graph,
-                 with_gradual_search: bool = False) -> nx.Graph:
-    original_polygon = polygon
-    polygon = add_sites(orient(polygon), sites=sites)
-    if edge is not None:
-        polygon = order_by_edge(polygon, edge=edge)
-    else:
-        polygon = order_by_site(polygon, sites=sites)
-    graph = nx.relabel_nodes(graph, {original_polygon: polygon})
-    if with_gradual_search and len(sites) == 1:
-        partitions_to_process = gradual_search(polygon=polygon,
-                                               sites=sites,
-                                               edge=edge)
-    else:
-        partitions_to_process, graph, neighbor = nonconvex_divide(polygon,
-                                                                  sites,
-                                                                  graph=graph)
-    partitions_union = unary_union([partition.polygon
-                                    for partition in partitions_to_process])
-    nodes_to_remove = [node for node in graph
-                       if partitions_union.contains(node.buffer(-1e-9))]
-    graph.remove_nodes_from(nodes_to_remove)
-    if len(partitions_to_process) == 3:  # Case 1.3
-        return prepend_partitions_inline(partitions_to_process,
-                                         graph=graph,
-                                         node_to_attach_to=neighbor)
-    else:  # Cases 1.1 and 1.2
-        return prepend_partitions(partitions_to_process,
-                                  graph=graph,
-                                  node_to_attach_to=neighbor)
+    ordered_points = sorted(points.points, key=angle)
+    site_index = next(index for index in range(len(ordered_points))
+                      if ordered_points[index] == site_location)
+    return ordered_points[site_index + 1:] + ordered_points[:site_index + 1]

```

Comment:
this can be extracted into separate one-liner known as `rotate`
```python
def rotate(sequence: Sequence[Domain], index: int) -> Sequence[Domain]:
    return sequence[index:] + sequence[:index]
```
and after that used here like
```python
return rotate(ordered_points, site_index + 1)
```

Reply:
it's a shame that Python built-ins has no such useful functionality